services/checks/checks.go

```diff
@@ -451,8 +452,36 @@ func (s *Service) minPMMAgentVersion(t check.Type) *version.Parsed {
 	}
 }
 
-// executeChecks runs all available checks for all reachable services.
-func (s *Service) executeChecks(ctx context.Context) error {
+func (s *Service) filterChecks(checks []check.Check, disable, enable []string) []check.Check {
+	var res []check.Check
+	disableMap := make(map[string]struct{}, len(disable))
+	for _, e := range disable {
+		disableMap[e] = struct{}{}
+	}
+
+	enableMap := make(map[string]struct{}, len(enable))
+	for _, e := range enable {
+		enableMap[e] = struct{}{}
+	}
+
+	for _, c := range checks {
+		if _, ok := disableMap[c.Name]; ok {
```

Comment:
should we show some error message or log if check was disabled in settings, but requested to run in API call?

Reply:
I think we shouldn't ignore that silently

Reply:
The whole function was rewritten during merge x_X